packages/teams-js/src/private/externalAppCardActionsForDA.ts

```diff
@@ -0,0 +1,121 @@
+/**
+ * @hidden
+ * Module to delegate adaptive card action execution to the host
+ * @internal
+ * Limited to Microsoft-internal use
+ * @module
+ */
+
+import { callFunctionInHost } from '../internal/communication';
+import { validateAppIdInstance, validateStringIdInstance } from '../internal/idValidation';
+import { ensureInitialized } from '../internal/internalAPIs';
+import { ApiName, ApiVersionNumber, getApiVersionTag } from '../internal/telemetry';
+import { DialogSize } from '../public';
+import { AppId } from '../public';
+import { errorNotSupportedOnPlatform, FrameContexts } from '../public/constants';
+import { runtime } from '../public/runtime';
+import { ValidatedStringId } from '../public/validatedStringId';
+import { isExternalAppError } from './externalAppErrorHandling';
+
+/**
+ * All of APIs in this capability file should send out API version v2 ONLY
+ */
+const externalAppCardActionsForDATelemetryVersionNumber: ApiVersionNumber = ApiVersionNumber.V_2;
+
+/**
+ * @beta
+ * @hidden
+ * Base interface for Dialog Information.
+ * @internal
+ * Limited to Microsoft-internal use
+ *
+ * @param title The title of the dialog.
+ * @param size The size of the dialog.
+ */
+export interface IDialogActionBase {
+  title: string;
+  size: DialogSize;
+}
+
+/**
+ * @beta
+ * @hidden
+ * Interface to define the Dialog info for Adaptive Card Action.OpenUrlDialog request.
+ * @internal
+ * Limited to Microsoft-internal use
+ *
+ * @param url The URL to open in the dialog.
+ */
+export interface IActionOpenUrlDialogInfo extends IDialogActionBase {
+  url: URL;
+}
+
+/**
+ * @beta
+ * @hidden
+ * Delegates an Adaptive Card Action.OpenUrlDialog request to the host for the application with the provided app ID
+ * @internal
+ * Limited to Microsoft-internal use
+ * @param appId ID of the application the request is intended for. This must be a UUID
+ * @param actionOpenUrlDialogInfo Information required to open the URL dialog
+ * @param traceId The trace identifier used for monitoring and live site investigations
+ * @returns Promise that resolves when the request is completed and rejects with ExternalAppError if the request fails
+ */
+export async function processActionOpenUrlDialog(
+  appId: AppId,
+  actionOpenUrlDialogInfo: IActionOpenUrlDialogInfo,
+  traceId: ValidatedStringId,
+): Promise<void> {
+  ensureInitialized(runtime, FrameContexts.content);
+
+  if (!isSupported()) {
+    throw errorNotSupportedOnPlatform;
+  }
+  validateInput(appId, traceId);
+
+  return callFunctionInHost(
+    ApiName.ExternalAppCardActionsForDA_ProcessActionOpenUrlDialog,
+    [
+      appId,
+      actionOpenUrlDialogInfo.url.href,
```

Comment:
https://office.visualstudio.com/DefaultCollection/ISS/_git/metaos-hub-sdk/pullRequest/3873862#1742253459

Reply:
We still need to serialize URL. What about this:
```
export class SerializableActionActionOpenUrlDialogInfo implements ISerializable {
  public constructor(private actionOpenUrlDialogInfo: IActionOpenUrlDialogInfo) {}

  public serialize(): object | string {
    return {
      url: this.actionOpenUrlDialogInfo.url.href,
      title: this.actionOpenUrlDialogInfo.title,
      size: this.actionOpenUrlDialogInfo.size,
    };
  }
}
```

Reply:
yup. it is good. my only concern is not have the parameters outside the object.